src/Security/Authentication/test/JwtBearerTests_Handler.cs

```diff
@@ -0,0 +1,1293 @@
+                            await context.ForbidAsync("JwtAuthSchemaOne");
+                            await context.ForbidAsync("JwtAuthSchemaTwo");
+                        });
+                    })
+                    .ConfigureServices(services =>
+                    {
+                        services.AddAuthentication()
+                                .AddJwtBearer("JwtAuthSchemaOne", o => { o.Events = jwtBearerEvents; })
+                                .AddJwtBearer("JwtAuthSchemaTwo", o => { o.Events = jwtBearerEvents; });
+                    }))
+            .Build();
+
+        await host.StartAsync();
+
+        using var server = host.GetTestServer();
+        var response = await server.CreateClient().SendAsync(new HttpRequestMessage(HttpMethod.Get, string.Empty));
+
+        Assert.Equal(418, (int)response.StatusCode);
+        Assert.Equal("You Shall Not Pass", await response.Content.ReadAsStringAsync());
+        Assert.Equal(2, onForbiddenCallCount);
+    }
+
+    [Fact]
+    public async Task ExpirationAndIssuedSetOnAuthenticateResult()
+    {
+        var key = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(new string('a', 128)));
+        var creds = new SigningCredentials(key, SecurityAlgorithms.HmacSha256);
+
+        var claims = new[]
+        {
+                new Claim(ClaimTypes.NameIdentifier, "Bob")
+            };
+
+        var token = new JwtSecurityToken(
+            issuer: "issuer.contoso.com",
+            audience: "audience.contoso.com",
+            claims: claims,
+            notBefore: DateTime.Now.AddMinutes(-10),
+            expires: DateTime.Now.AddMinutes(30),
+            signingCredentials: creds);
+
+        var tokenText = new JwtSecurityTokenHandler().WriteToken(token);
+
+        using var host = await CreateHost(o =>
+        {
+            o.SaveToken = true;
+            o.TokenValidationParameters = new TokenValidationParameters()
+            {
+                ValidIssuer = "issuer.contoso.com",
+                ValidAudience = "audience.contoso.com",
+                IssuerSigningKey = key,
+            };
+        });
+
+        var newBearerToken = "Bearer " + tokenText;
+        using var server = host.GetTestServer();
+        var response = await SendAsync(server, "http://example.com/expiration", newBearerToken);
+        Assert.Equal(HttpStatusCode.OK, response.Response.StatusCode);
+        var responseBody = await response.Response.Content.ReadAsStringAsync();
+        using var dom = JsonDocument.Parse(responseBody);
+        Assert.NotEqual(DateTimeOffset.MinValue, token.ValidTo);
+        Assert.NotEqual(DateTimeOffset.MinValue, token.ValidFrom);
+        Assert.Equal(token.ValidTo, dom.RootElement.GetProperty("expires").GetDateTimeOffset());
+        Assert.Equal(token.ValidFrom, dom.RootElement.GetProperty("issued").GetDateTimeOffset());
+    }
+
+    [Fact]
+    public async Task ExpirationAndIssuedWhenMinOrMaxValue()
```

Comment:
Note: behavior is slightly different than:

https://github.com/dotnet/aspnetcore/blob/d9bf9d3274992e9e7d64ff6f902ec48b74ff2f00/src/Security/Authentication/test/JwtBearerTests.cs#L933

As JsonWebToken correctly handles datetimes represented by greater than int32 values